### ISSUE TITLE
Keybind on release

### ISSUE THREAD

**Issue by Akinevz** [herbstluftwm/herbstluftwm#967]
There doesn't appear to be any functionality to run a keybind on key-release. 

Is it against herbstluftwm's philosophy to start performing an action on keypress and stop on release?
Such as - resize window while arrow held

**Comment by t-wissmann** [comment]
Having keybinds on key release is on the todo-list in my head and I'm wondering whether there is no github issue for this :)

**Comment by Akinevz** [comment]
Could you link me the file where keybind are implemented? I'll have a look and send a patch, sounds like good plans for next week

**Comment by t-wissmann** [comment]
The keybinding stuff is implemented in `src/keymanager.cpp` and `src/xkeygrabber.cpp`.

**Comment by schieni** [comment]
By the way: Due to the Client/Server design, wouldn't it be better to use an external input handler like sxhkd and remove the corresponding project part? Perhaps forking one, to better fits the needs?

**Comment by t-wissmann** [comment]
> Due to the Client/Server design, wouldn't it be better to use an external input handler like sxhkd and remove the corresponding project part? Perhaps forking one, to better fits the needs?
> 
> Greetings from Wuppertal, Germany.

:-)

The reason I don't like external key handling is that a new process needs to be spawned on each and every keypress. With the keybinding handling directly in hlwm, the hlwm commands are directly execued without any need of invoking 'herbstclient'. The architecture of hlwm is that you need this client/server architecture to wire things together at config time but at run-time everything (or at least a big amount) happens only in the 'server' process (a bit like `iptables`: here you call iptables multiple times to setup the network rules, but no new processes are invoked while the network rules are executed and the network packages are routed around)

**Comment by schieni** [comment]
Ok, this is a valid point. Thanks!

**Comment by Akinevz** [comment]
Are there any numbers on whether it is faster to have a wm+keybind
server/daemon vs a separate daemon running foreground in the wm, sending
IPC to the client?

**Comment by t-wissmann** [comment]
No, I don't have benchmarks on this. But beside unnecessary resource usage, I don't like it because it's fragile: if you attempt to recompile/upgrade hlwm and the herbstclient executable is gone (or just not in the $PATH), then all your keybinds stop involving 'herbstclient' will stop working. If for some reason. Similarly, if your system has heavy memory it might be reluctant to fork new processes, and then focusing another window will take forever because it involves forking a new herbstclient process.
